### PR TITLE
chore: migrate from tap to node:test and c8

### PR DESCRIPTION
**Pull request by dancastillo**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,18 +27,18 @@
     "lint:everything": "npm run lint && npm run test:types",
     "lint:fix": "standard --fix",
     "lint:standard": "standard --verbose | snazzy",
-    "test:mocha": "tap",
+    "test:unit": "c8 --statements 98 --branches 97 --functions 96 --lines 98 node --test",
```
Comment (dancastillo) [inline package.json:30]: cannot do `c8 ---100` since there is not already 100% coverage being tracked here https://github.com/fastify/busboy/issues/126

**Review by gurgunday — APPROVED** [review]

**Comment by dancastillo** [comment]
@gurgunday do you have permission to merge this if its good to go? I dont have permission on this repo

**Review by gurgunday — APPROVED** [review]
lgtm

File: package.json (hunk 1 above)
Comment (gurgunday) [reply package.json:30]: nice I didn't know we could separate them

**Review by gurgunday — APPROVED** [review]

**Pull request merged**